tekton/resources/cd/eventlistener.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright 2020 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: tekton-cd
+spec:
+  serviceAccountName: tektoncd
+  triggers:
+    - name: configmaps
+      interceptors:
+        - cel:
+            filter: >-
+              'configmap' in body
+      bindings:
+        - name: deploy-target-details
+        - name: deploy-source-git
+        - name: configmap-details
+      template:
+        name: deploy-configmap
+    - name: folders
+      interceptors:
+        - cel:
+            filter: >-
+              'folder' in body
```

Comment:
could you explain more about what these filters are looking for (maybe in a comment?)

im a bit confused what the contents of the body would be here if we're triggering with a cron and why sometimes they'd contain configmap, sometimes folder, and why thats enough info to filter on

Reply:
This is a "fan out" that is used by the event listener to trigger the right type of deployment.
It basically allows having one entry point for deployment jobs, which can be used to deploy config maps, or resources from a folder or other types in future, based on the parameters sent in the body.

The alternative would be to have multiple event listeners - which may be a valid one.
It seemed to me that having one running service only reduces a bit boiler plate code and makes it more manageable, but I don't have a very strong feeling about this.